cloud-aws/src/main/resources/templates/aws-cf-stack.ftl

```diff
@@ -515,15 +558,24 @@
     }
   </#if>
   <#if !existingVPC>
-    ,"CreatedVpc": {
+    <#if mapPublicIpOnLaunch || (enableInstanceProfile && !existingRole)>,</#if>
+    "CreatedVpc": {
         "Value" : { "Ref" : "VPC" }
     }
   </#if>
   <#if !existingSubnet>
-    ,"CreatedSubnet": {
+    <#if mapPublicIpOnLaunch || (enableInstanceProfile && !existingRole) || !existingVPC>,</#if>
```

Comment:
Yeah, I can't quite follow the changes you made to these if statements around the network settings. Can you talk through them?

Reply:
the condition is when to add "," here. If there is no more element, adding "," will result in invalid template. If there is element, not adding "," will result in invalid template. 

This change makes sure that the template is always valid in different test scenarios. 

I have encountered failure, so I have to make such change to get all tests pass.